Modules/Loadable/Markups/MRML/vtkCurveMeasurementsCalculator.cxx

```diff
@@ -314,6 +326,176 @@ bool vtkCurveMeasurementsCalculator::CalculatePolyDataCurvature(vtkPolyData* pol
   return true;
 }
 
+//------------------------------------------------------------------------------
+bool vtkCurveMeasurementsCalculator::CalculatePolyDataTorsion(vtkPolyData* polyData)
+{
+  if (polyData == nullptr)
+    {
+    return false;
+    }
+  if (polyData->GetNumberOfPoints() == 0 || polyData->GetNumberOfLines() == 0)
+    {
+    return false;
+    }
+
+  if (this->Measurements->GetNumberOfItems() == 0)
+    {
+    vtkErrorMacro("No measurements found");
+    return false;
+    }
+  vtkMRMLMeasurement* firstMeasurement = vtkMRMLMeasurement::SafeDownCast(this->Measurements->GetItemAsObject(0));
```

Comment:
Shouldn't we look for torsion measurements specifically instead of just the first measurement? If there are no torsion measurements enabled then we may skip the torsion computation completely.

Also, if we need the markup node anyway then it may be cleaner to set the markup node (instead of the measurements object) as input in vtkCurveMeasurementsCalculator.

Reply:
The first measurement at this specific line was used just for getting the node. However, it's indeed cleaner to set/get the MRML node instead, I changed the code accordingly.